### ISSUE TITLE
Why using max(n, maxV) as n input to dct function?

### ISSUE THREAD

**Issue by germunwang** [chalmersgit/Discrete-Cosine-Transform#1]
Hi, 
I have a question at
https://github.com/chalmersgit/Discrete-Cosine-Transform/blob/43b37de3b6c5211d54a8254406653268ebb6c95b/dct.py#L51

`                    basisImg[y,x] = dct(x, y, v, u, max(n, maxV))`

Why are you using `max(n, maxV)` instead of `n` as argument input `n`?
All related formula I found are using range N of (x,y) but not range K of (u,v), is there some reason to use `max()` of them? (although in generally N=K)

I think that may be because of alpha function, but should it be use to normalize part of cosine x and y too?

Also, it seems not a sense to normalize `cosine y` part using `maxV`
Or should we use two different argument to normalize in alpha and cosine part respectively?

Thanks a lot.

**Comment by chalmersgit** [comment]
Hi,

It's been a while since I've looked at this, and I'm not sure what my thinking was when adding the max function. I think I must've been adjusting the normlisation for varying x and y resolution, but as you pointed out, it doesn't make sense since I didn't account for it on the other axis.

To answer your second question, yes I believe that is correct. You would have two different arguments to normalize each axis respectively (the alpha function would use width/height instead of n). I've updated my code to remove the ambiguity. It takes n as an input argument and assumes a square image. 

Hope that clears thing up,
Thanks.
